tidy3d/plugins/autograd/differential_operators.py

```diff
@@ -1,17 +1,53 @@
 from typing import Any, Callable
 
+from autograd import grad as grad_ag
 from autograd import value_and_grad as value_and_grad_ag
 from autograd.builtins import tuple as atuple
 from autograd.core import make_vjp
 from autograd.extend import vspace
 from autograd.wrap_util import unary_to_nary
 from numpy.typing import ArrayLike
 
+from .utilities import scalar_objective
+
 __all__ = [
     "value_and_grad",
+    "grad",
 ]
 
 
+def grad(fun: Callable, has_aux: bool = False) -> Callable:
+    """Returns a function that computes the gradient of the input function.
+
+    This function wraps autograd's 'grad' function by first wrapping the input
+    function in scalar_objective. If has_aux is True, it uses autograd's 'value_and_grad' function
+    and extracts the gradient and auxiliary data.
+
+    Parameters
+    ----------
+    fun : Callable
+        The function to differentiate. Should return a scalar value.
+    has_aux : bool = False
+        If True, the function returns auxiliary data as the second element of a tuple.
+
+    Returns
+    -------
+    Callable
+        A function that takes the same input as `fun` and returns the gradient.
+    """
+    wrapped_fun = scalar_objective(fun, has_aux=has_aux)
+
+    if has_aux:
+
+        def grad_with_aux(*args: Any, **kwargs: Any) -> tuple:
+            (_, grad_value), aux = value_and_grad(wrapped_fun, has_aux=has_aux)(*args, **kwargs)
+            return grad_value, aux
+
+        return grad_with_aux
+    else:
+        return grad_ag(wrapped_fun)
+
+
 @unary_to_nary
 def value_and_grad(
     fun: Callable, x: ArrayLike, *, has_aux: bool = False
```

Comment:
do we want to pass `**kwargs` to `scalar_objective`?

comment applies to scalar objective and `grad` too.

Reply:
not really sure what you mean here? `grad_ag(wrapped_fun)` is still a callable to which you have to pass all function arguments. or am I misunderstanding?

Reply:
what I mean is that it seems like the other kwargs passable to `value_and_grad` and `grad` are not supported with the autograd wrapper here? is that right?

```
def grad(fun, argnum=0, *nary_op_args, **nary_op_kwargs)
```

Reply:
oooh yeah good point.. yeah this looks a bit tricky I'll try tomorrow